src/Infrastructure.VS/Roslyn/IRoslynWorkspaceWrapper.cs

```diff
@@ -36,20 +37,27 @@ internal interface IRoslynWorkspaceWrapper
 internal class RoslynWorkspaceWrapper : IRoslynWorkspaceWrapper
 {
     private readonly Workspace workspace;
+    private readonly IThreadHandling threadHandling;
 
     [ImportingConstructor]
     [ExcludeFromCodeCoverage] // not mef-testable
-    public RoslynWorkspaceWrapper(VisualStudioWorkspace workspace) : this(workspace as Workspace)
+    public RoslynWorkspaceWrapper(VisualStudioWorkspace workspace) : this(workspace as Workspace, ThreadHandling.Instance)
     {
     }
 
-    internal /* for testing */ RoslynWorkspaceWrapper(Workspace workspace)
+    internal /* for testing */ RoslynWorkspaceWrapper(Workspace workspace, IThreadHandling threadHandling)
     {
         this.workspace = workspace;
+        this.threadHandling = threadHandling;
     }
 
     public IRoslynSolutionWrapper CurrentSolution => 
         new RoslynSolutionWrapper(workspace.CurrentSolution);
```

Comment:
Can .CurrentSolution be accessed on a background thread?

Reply:
Yes. The only problem was when the TryApplyChanges is called on a background thread